src/main.cpp

```diff
@@ -1492,6 +1492,20 @@ class ATTR_DLL_LOCAL FragmentedSampleReader : public SampleReader, public AP4_Li
                            AP4_CencSampleDecrypter::Create(sample_table, algorithm_id, 0, 0, 0, reset_iv,
                                                            m_singleSampleDecryptor, m_decrypter)))
           return result;
+
+        // Inform decrypter of pattern decryption (CBCS)
+        AP4_UI08 cryptBlocks = sample_table->GetCryptByteBlock();
+        AP4_UI08 skipBlocks = sample_table->GetSkipByteBlock();
+
+        if (cryptBlocks > 0 && skipBlocks > 0)
```

Comment:
Just a question, i see that here you give priority to these variables to set CBCS,
is because `m_protectedDesc->GetSchemeType()` not provide the right scheme?

Reply:
Audio will come back with 0,0 for these values (which is 100% of samples encrypted) and video will usually be 1,9. 

I could rearrange it to a 
```
if (CBCS) 
{
  if (blocks >0)
  {
   //send pattern and cbcs
  }
else
  {
  // just send cbcs
  }
}
else
{
// send cenc
}
```
if you wanted with the same result?

Reply:
if it is the same yes seem more correct

Reply:
Ok I think it's much better now